File: v1/src/main/java/com/api/v1/medical_slots/services/MedicalSlotRetrievalService.java
package com.api.v1.medical_slots.services;

import com.api.v1.medical_slots.DefaultMedicalSlotResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;

public interface MedicalSlotRetrievalService {
    ResponseEntity<DefaultMedicalSlotResponseDTO> findByIdAndDoctor(String id, String medicalLicenseNumber);
    ResponseEntity<Page<DefaultMedicalSlotResponseDTO>> findAllByDoctor(String medicalLicenseNumber, Pageable pageable);
    ResponseEntity<Page<DefaultMedicalSlotResponseDTO>> findAll(Pageable pageable);
}

File: v2/src/main/kotlin/com/api/v2/medical_slots/services/MedicalSlotRetrievalService.kt
package com.api.v2.medical_slots.services

import com.api.v2.medical_slots.DefaultMedicalSlotResponseDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity

interface MedicalSlotRetrievalService {
    fun findByIdAndDoctor(id: String, medicalLicenseNumber: String): ResponseEntity<DefaultMedicalSlotResponseDTO>
    fun findAllByDoctor(medicalLicenseNumber: String, pageable: Pageable): ResponseEntity<Page<DefaultMedicalSlotResponseDTO>>
    fun findAll(pageable: Pageable): ResponseEntity<Page<DefaultMedicalSlotResponseDTO>>
}